src/components/date/date.component.js

```diff
@@ -141,13 +141,10 @@ class BaseDateInput extends React.Component {
       this.reformatVisibleDate();
 
       if (this.props.onBlur && !this.state.isDatePickerOpen) {
-        const dateWithSlashes = DateHelper.sanitizeDateInput(
-          this.state.visibleValue
-        );
         const event = this.buildCustomEvent(
           { target: this.input, type: "blur" },
           DateHelper.formatValue({
-            value: dateWithSlashes,
+            value: this.state.visibleValue,
```

Comment:
I think this will prevent the date from auto formatting on blur, it will just leave the value in the input as typed which is not the correct behaviour

Reply:
we do not need to change the value's delimiters to slashes and then change them back again to the proper value, as the format method does tat anyway. By changing them to slashes we were introducing an issue, in which when an user did not specified a date containing slashes in the list of valid formats, an error would be thrown, as moment would compare the incorrect slashes date instead of what user typed in.

Reply:
This is not how the date should behave, IMHO it needs to reformat to the locale based format when the input blurs, currently it doesn't set the picker or reformat and then on refocusing the picker it jumps back to different date

How it behaves in master:
https://user-images.githubusercontent.com/44157880/142845573-6970d2ec-88c2-4c2b-9af2-65c216cff38c.mov

After your changes:
https://user-images.githubusercontent.com/44157880/142845628-c8275b3c-bdc3-4483-92c6-0d557ece38dd.mov

Reply:
This is not how the date should behave, IMHO it needs to reformat to the locale based format when the input blurs, currently it doesn't set the picker or reformat and then on refocusing the picker it jumps back to different date

How it behaves in master:
![2021-11-22 10 30 29](https://user-images.githubusercontent.com/44157880/142845845-ac2484d0-ba82-47a7-a682-9b8a38e69d44.gif)

After your changes:
![2021-11-22 10 31 17](https://user-images.githubusercontent.com/44157880/142846011-ac8c49a9-80ea-4ebf-801a-2c05bd7a49f6.gif)

Reply:
looks like it was already broken, but we were enforcing it to have slashes, which shouldn't be done. Good spot!

Reply:
I'm taking that back, the date with dot delimiter is not a valid date, as it's not declared in the valid date list